src/csharp/Grpc.HealthCheck.Tests/HealthServiceImplTest.cs

```diff
@@ -83,6 +84,43 @@ public void NullsRejected()
             Assert.Throws(typeof(ArgumentNullException), () => impl.ClearStatus(null));
         }
 
+#if GRPC_SUPPORT_WATCH
+        [Test]
+        public async Task Watch()
+        {
+            var cts = new CancellationTokenSource();
+            var context = new TestServerCallContext(cts.Token);
+            var writer = new TestResponseStreamWriter();
+
+            var impl = new HealthServiceImpl();
+            var callTask = impl.Watch(new HealthCheckRequest { Service = "" }, writer, context);
+
+            var nextWriteTask = writer.WaitNextAsync();
+            impl.SetStatus("", HealthCheckResponse.Types.ServingStatus.Serving);
```

Comment:
also test for other service name than `""`? (e.g. `grpc.test.TestService`)

Reply:
Done